src/Illuminate/Testing/Fluent/Concerns/Matching.php

```diff
@@ -169,7 +162,7 @@ public function whereContains(string $key, $expected)
         );
 
         $missing = Collection::make($expected)
-            ->map(fn ($search) => $search instanceof BackedEnum ? $search->value : $search)
+            ->map(fn ($search) => mutate($search))
```

Comment:
This could probably be shortened to:
```suggestion
       $missing = Collection::make($expected)
            ->map(mutate(...))
```

Reply:
`[Foo::Bar, null]` will become `['bar', 1]` with the above changes instead of `['bar', null]`